### PR TITLE
qt: Redesign of the main toolbar

### PR DESCRIPTION
**Pull request by xdustinface**
This PR ist part of a series of +-25 PRs related to UI redesigns. Its ancestor is #3508, its successor is #3555. I did not screenshot every single PR and its changes, instead i made "walk through all screen" videos with the result of this PR series and also with the 0.15 UI. If there are any concrete screenshots wanted, just let me know. To build with the full set of changes you can build from the branch [xdustinface:pr-ui-redesign](https://github.com/xdustinface/dash/tree/pr-ui-redesign) which always contains all changes.

[ -> Walk through 0.15](https://youtu.be/cWQeHj5tWR0)
[ -> Walk through Redesign](https://youtu.be/0QeSyXo1aao)

I tried to give the commits enough text to make things obvious without a lot description for each PR. Also here, if you want more description for this specific PR, let me know.
### About this PR

This PR gives the toolbar state related styles/colors and stretches its buttons over the window's width. It also brings a dash logo with higher resolution to the toolbar.

All PRs are based on its ancestor. [Click here to see the changes of this PR only](https://github.com/xdustinface/dash/compare/pr-ui-css-generalization...xdustinface:pr-ui-1-toolbar)

### CODE REVIEW

**Comment by strophy** [comment]
This looks fantastic! What a difference! Just one question, has this been tested under Windows 10 with a HiDPI display panel and scaling enabled? See #3311 and #3344 and let me know if you want me to do any testing on my hardware, since I have had scaling and/or blurriness problems for several versions now.

**Comment by xdustinface** [comment]
> This looks fantastic! 

Thanks! 

> Just one question, has this been tested under Windows 10 with a HiDPI display panel and scaling enabled? 

Kind if yeah. Its developed on macOS but i did test with windows/linux in a VM. Tried a bit with the scaling because i had those two issue in mind but still, the more different testing environments the better, so sure, please do your tests 🙂 

Related to the blurriness you will still find issues probably at least on splashscreen and the address in the QR code image. Its on my list to investigate though!

File: src/qt/res/css/trad.css [hunk 1]
```diff
@@ -25,6 +25,43 @@ Loaded in GUIUtil::loadStyleSheet() in guitil.cpp.
...
+
+QToolBar > QToolButton:checked {
+    background-color: #00000000;
+    color: #333;
+}
+
+QToolBar > QToolButton:disabled {
+    color: #444;
+}
+
```
Comment (UdjinM6) [inline src/qt/res/css/trad.css:64]: I'd keep styling for the Traditional scheme minimal i.e.
```suggestion

```

Screenshot 2020-06-27 at 14 16 43

Comment (PastaPastaPasta) [reply src/qt/res/css/trad.css:64]: I agree with keeping traditional limited

**Comment by xdustinface** [comment]
Sure, there is nothing could say against keeping the old styles there 😄 

Added the suggestions + removed the blue background from the label, re-added the blue logo with a higher resolution and put it into the traditional toolbars logo label. Looks like:

Bildschirmfoto 2020-06-27 um 22 17 22

**Comment by strophy** [comment]
A few notes after testing on native HiDPI Windows 10:

- Blurriness remains on splash screen, but is fixed everywhere else as far as I can tell. The QR code looks fine, but the text address below it is blurry. All other blurriness is gone - this was my main concern and I'm thrilled it is fixed :)
- I find reading Montserrat extremely uncomfortable for body text because of the character width. Montserrat is really designed for headings, not body text - can we default to the system default font as body font for consistency? Or offer Open Sans instead/as well as Montserrat?
- Changing from Montserrat to SystemDefault results in serious kerning issues in the default font, see [here](https://imgur.com/a/DTzkwzg) for screenshot. Dragging the window from a HiDPI to regular DPI screen results in the window bar appearing double thickness and height, but the kerning problems in the rest of the application disappear. I think most of this scaling, smoothing and kerning is handled by ClearType in other apps, is this not the case here?
- Send screen: "Subtract fee from amount" checkbox touching amount box
- Every screen has some buttons where descenders on "y" or "g" are vertically truncated, and some horizontal truncation of long text on buttons as well
- Most settings in the font weight slider have no effect if system font does not support that many weights

In general I find that even when using the scaling sliders, it is impossible to get the same size and weight of the font in the menu bar. So the application always feels like it "sticks out" and goes against the grain of other Windows apps because the text is a different size, weight, font and scaling than every other app on my computer. Is it possible to get an option to just use the system settings for everything, so it feels more "native"?

**Comment by UdjinM6** [comment]
@strophy your comment looks unrelated to this specific PR. I created a separate issue to make sure it's not lost after this PR is merged. Pls move the discussion about these and other issues there (or to specific PRs) instead.

**Pull request merged**